crypto/ring-signature/src/ring_signature/mlsag.rs

```diff
@@ -270,7 +270,7 @@ mod mlsag_tests {
             };
 
             let real_index = rng.next_u64() as usize % (num_mixins + 1);
-            ring.insert(real_index, reduced_tx_out);
+            let _ = ring.insert(real_index, reduced_tx_out);
```

Comment:
I'm a bit confused by this, these are heapless errors that we were ignoring before?

I think maybe it's better to panic rather than swallowing these errors?

Reply:
yeah updated clippy lint was giving me failures... i'll go back and make sure this is the case but these errors _should_ be unreachable because we fail on extending the vec to the required ring length earlier in the process. prefer a panic or a propagated error?

Reply:
if it's like, a numeric limit is exceeded, and we should know statically that it can't be reached, then a panic seems correct, since it's a logic error?

if possible, we could pair that with like a `static_assert`? (I haven't read all the surrounding code)

Reply:
if you really think it's unreachable then panic seems good

Reply:
dropped these changes for the moment

Reply:
Let's make a github issue or something to track it?